nbdime/args.py

```diff
@@ -0,0 +1,95 @@
+# coding: utf-8
+
+# Copyright (c) IPython Development Team.
+# Distributed under the terms of the Modified BSD License.
+
+import os
+
+from ._version import __version__
+
+
+def add_generic_args(parser):
+    """Adds a set of arguments common to all nbdime commands.
+    """
+    parser.add_argument(
+        '--version',
+        action="version",
+        version="%(prog)s " + __version__)
+    if 0:  # TODO: Use verbose and quiet across nbdime and enable these:
+        qv_group = parser.add_mutually_exclusive_group()
+        qv_group.add_argument(
+            '-v', '--verbose',
+            default=False,
+            action="store_true",
+            help="increase verbosity of console output.")
+        qv_group.add_argument(
+            '-q', '--quiet',
+            default=False,
+            action="store_true",
+            help="silence console output.")
+
+
+def add_web_args(parser, default_port=8888):
+    """Adds a set of arguments common to all commands that show a web gui.
+    """
+    port_help = ("specify the port you want the server "
+                "to run on. Default is %d%s." % (
+                    default_port,
+                    " (random)" if default_port == 0 else ""
+    ))
+    parser.add_argument(
+        '-p', '--port',
+        default=default_port,
+        type=int,
+        help=port_help)
+    cwd = os.path.abspath(os.path.curdir)
+    parser.add_argument(
+        '-w', '--workdirectory',
+        default=cwd,  # TODO: Are there any security implications of doing this?
```

Comment:
This todo came along with a cut and paste, so I'm not the one asking the question. As I am not certain about the answer though, I left it untouched.